pkgs/build-support/writers/test.nix

```diff
@@ -8,9 +8,46 @@
 , runCommand
 , writers
 , writeText
+# for `trivial` tests
+, callPackage
+, writeShellScript
+# makes it possible to disable tests to save local rebuilding while testing a specific set
+# interface:
+#  the `enabledTest` function returns the test if a true entry exists, defaults to false otherwise. (for convenience)
+#  The attrnames by convention correspond to the names in the large let expression:
+#  as used by the runCommand at the end of the file.
+, enabledTests ? { trivial = true; bin = true; simple = true; path = true; }
 }:
 with writers;
 let
+  # These aren't really writers tests, but I wouldn't know where else to put them.
+  trivial = let
+    bash_script =  ''
+      shopt -s extglob
+      echo @(foo|bar)
+    '';
+    check_bash = old: ''
+      export BASHOPTS
+      shopt -s extglob
+      ${old.checkPhase}
+      # Technically, if you want to restore the previous state, you want to save
+      # the previous state and restore it, instead of set/unset + export/unexport;
+      # this is just meant as a simple example.
+      shopt -u extglob
+      export -n BASHOPTS
+    '';
+    assert_fail = old: ''
+      ! { ${old.checkPhase}; }
+    '';
+    plain = case: writeShellScript "test-trivial-plain-${case}" bash_script;
+    callp = case: callPackage ({writeShellScript}: writeShellScript "test-trivial-callp-${case}" bash_script) {};
```

Comment:
I don't think this one isn't necessary anymore since `callPackage` does *only* screw with `override` and if something else happened to `overrideAttrs`, the tests would fail anyways.